### PR TITLE
Add neuronx-nemo-megatron training usecase

### PR DESCRIPTION
**Pull request by KeitaW**
*Issue #, if available:*

*Description of changes:*
This PR introduces Llama2 training example with neuronx-nemo-megatroin on Trn1.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: 3.test_cases/8.neuronx-nemo-megatron/README.md [hunk 1]
```diff
@@ -0,0 +1,154 @@
+# NeuronX Nemo Megatron Llama2 trainining on Trn1 Test Case <!-- omit in toc -->
+
+[AWS Neuron Reference for NeMo Megatron](https://github.com/aws-neuron/neuronx-nemo-megatron/tree/main)(`neuronx-nemo-megatron`) is a modified versions of open-source packages [Nemo](https://github.com/NVIDIA/NeMo) and [Apex](https://github.com/NVIDIA/apex) that have been adapted for use with [AWS Neuron](https://awsdocs-neuron.readthedocs-hosted.com/) and [Amazon EC2 Trn1 instance](https://aws.amazon.com/jp/ec2/instance-types/trn1/). This test case describes how to run Llama2 training on Slurm with Trn1 instances.
```
Comment (verdimrc) [inline 3.test_cases/8.neuronx-nemo-megatron/README.md:3]: https://aws.amazon.com/jp/ec2/instance-types/trn1/ => change to en?

File: 3.test_cases/8.neuronx-nemo-megatron/README.md [hunk 2]
```diff
@@ -0,0 +1,154 @@
...
+/fsx/Llama2-meta/
+├── 7B/
+│   ├── checklist.chk
+│   ├── consolidated.00.pth
+│   └── params.json
+├── tokenizer.model
+└── tokenizer_checklist.chk
+```
+
+To conver the model to the standard Hugging Face format, the following script in transformers can be called with the following (example) command:
```
Comment (verdimrc) [inline 3.test_cases/8.neuronx-nemo-megatron/README.md:33]: fix typo: conver -> convert

Comment (verdimrc) [reply 3.test_cases/8.neuronx-nemo-megatron/README.md:33]: make transformers to url (or different font), as this is the 1st mention in this doc.

File: 3.test_cases/8.neuronx-nemo-megatron/README.md (hunk 1 above)
Comment (KeitaW) [reply 3.test_cases/8.neuronx-nemo-megatron/README.md]: changed thanks for catching this!

File: 3.test_cases/8.neuronx-nemo-megatron/README.md (hunk 2 above)
Comment (KeitaW) [reply 3.test_cases/8.neuronx-nemo-megatron/README.md]: fixed. ty!

File: 3.test_cases/8.neuronx-nemo-megatron/README.md [hunk 3]
```diff
@@ -0,0 +1,154 @@
...
+/fsx/Llama2-meta/
+├── 7B/
+│   ├── checklist.chk
+│   ├── consolidated.00.pth
+│   └── params.json
+├── tokenizer.model
+└── tokenizer_checklist.chk
+```
+
+To convert the model to the standard Hugging Face format, the following script in transformers can be called with the following (example) command:
```
Comment (KeitaW) [inline 3.test_cases/8.neuronx-nemo-megatron/README.md:33]: Here

File: 3.test_cases/8.neuronx-nemo-megatron/README.md [hunk 4]
```diff
@@ -0,0 +1,154 @@
+# NeuronX Nemo Megatron Llama2 trainining on Trn1 Test Case <!-- omit in toc -->
+
+[AWS Neuron Reference for NeMo Megatron](https://github.com/aws-neuron/neuronx-nemo-megatron/tree/main)(`neuronx-nemo-megatron`) is a modified versions of open-source packages [Nemo](https://github.com/NVIDIA/NeMo) and [Apex](https://github.com/NVIDIA/apex) that have been adapted for use with [AWS Neuron](https://awsdocs-neuron.readthedocs-hosted.com/) and [Amazon EC2 Trn1 instance](https://aws.amazon.com/ec2/instance-types/trn1/). This test case describes how to run Llama2 training on Slurm with Trn1 instances.
```
Comment (KeitaW) [inline 3.test_cases/8.neuronx-nemo-megatron/README.md:3]: and here corresponding to the revised lines.

File: 3.test_cases/8.neuronx-nemo-megatron/README.md (hunk 2 above)
Comment (KeitaW) [reply 3.test_cases/8.neuronx-nemo-megatron/README.md]: added link to the HF website.

**Comment by verdimrc** [comment]
LGTM

**Pull request merged**